engine/source/public/core/asset.hpp

```diff
@@ -0,0 +1,70 @@
+﻿#pragma once
+#include <string>
+
+namespace lumina
+{
+    enum class AssetType
+    {
+        Texture, // Including Samplers or not can be discussed
+        Shader,
+        Model, // Collections of meshes
+        Mesh,
+        Material,
+        None
+    };
+
+    class Asset
+    {
+        friend class AssetManager;
+
+    private:
+        static size_t nextID;
+
+    protected:
+        size_t id;
+        AssetType type;
+        std::string filePath;
+
+        Asset(AssetType type) : id(0), type(type) {}
+
+        virtual ~Asset();
+
+        virtual void Reload() {}
+
+        static size_t GenerateID(const std::string& path)
+        {
+            return std::hash<std::string> {}(path);
+        }
+
+        static size_t GetNextID()
+        {
+            return nextID++;
```

Comment:
In the asset_manager.cpp you used `++it` and here it's `nextID++`. it's a small consistency, but in this case it doesn't really matter whether we use pre- or post- increment. pre- is less steps to increment the variable, but I doubt it would make a difference for performance since it will probably be optimized by the compiler anyways.

Reply:
I might be a little bit faster, but I don't know what we want to do here

Reply:
Optimization-wise, it might be good to use pre- until it matters that we use a post-. I would also have to get used to that, since I usually only use post-increment

Reply:
So do I, but sometimes for integers it's better to pre-increment to avoid making a copy. The performance should be extremely small however.

Reply:
Performance would indeed be negligible. But it might be a good practise to write things as optimized as possible and not rely on the compiler. So if you agree, I would suggest we use pre- whenever possible